pallets/gear/src/internal.rs

```diff
@@ -572,27 +578,58 @@ where
                 }
 
                 // Cutting gas for storing in mailbox.
-                GasHandlerOf::<T>::cut(origin_msg, dispatch.id(), gas_limit)
+                GasHandlerOf::<T>::cut(msg_id, dispatch.id(), gas_limit)
                     .unwrap_or_else(|e| unreachable!("GasTree corrupted! {:?}", e));
+
+                if let Some(reservation_id) = reservation {
```

Comment:
`// TODO: issue #ref: adapt this line if gasful sending appears for reservations `

Reply:
Can you explain?

Reply:
Add line or mention in issue that this logic should be rewritten once sending wgas from reservation appears